### DISCUSSION TITLE
The cleanest way to change NvChad key mappings

### DISCUSSION THREAD

**Discussion by rfc2119** [NvChad/NvChad#746] [Q&A] [answered]
In `core/default_config.lua`, all key maps are defined as such:
```lua
   map("n", m.references, "<cmd>lua vim.lsp.buf.references()<CR>")
```
Since the key `m.references` is a variable, it can be modified later inside chadrc as follows:
```lua
local M = {}
M.mappings = {}
M.mappings.plugins = {
  lspconfig = {
    references = "C-k",
  }
}
```

However, the command itself is not configurable. What is the cleanest way to modify NvChad's key mapping ? For a quick fix, I modified `core/default_config.lua` although it's discouraged:
```lua
   map("n", m.references, "<cmd>Telescope lsp_references<CR>")
```

_P.S: If Telescope is installed by default, why not use its lsp_references instead of calling vim.lsp.buf.references() directly ?_

_Edit_: here's the answer based on @siduck  reply. Put this in your `custom/init.lua`:
```lua
local utils = require "core.utils"
local map = utils.map
local c = utils.load_config().mappings.plugins.lspconfig

map("n", c.references, "<cmd>Telescope lsp_references<CR>")
```

**Comment by siduck** [comment, marked as answer]
we use lspconfig's default keybinds! and also https://nvchad.github.io/config/Custom%20config#add-mappings for adding own mappings. To modify default mappings, you edit them by chadrc.

**Reply by rfc2119** [reply]
Great, thanks!

**Reply by pavelskipenes** [reply]
dead link. Updated https://nvchad.com/docs/config/mappings#add_new_mappings
